### PR TITLE
Convert cookbook to a 12.5 custom resource that is consumed by the default recipe

### LINKED ISSUE #107: Converge fails on windows when compile_time is true because 7z isn't available
### Cookbook version
3.20.0

### Chef-client version
12.9.41

### Platform Details
Windows 2012 R2 using azure rm driver

### Scenario:
Converge cookbook on windows 2012 R2 with compile_time set to true

### Steps to Reproduce:
Run test kitchen with compile_time set to true

### Expected Result:
cookbook converges

### Actual Result:
converge fails because 7z isn't installed

### PR DESCRIPTION
**Pull request by tas50**
### Description

This converts the cookbook into a compile time custom resource. 1) Does compile time the right way 2) Allows people to just say:

``` ruby
build_essential 'install the stuff'
```
### Issues Resolved

Potentially fixes #107 
### Check List
- [x] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

This just jams everything into a single resource, but that might actually make more sense.  Now that we're guaranteed to be on Chef 12 we can use multipackage too to speed things up

Potentially breaking changes:

Removes support for omnios 151008 which is EOL
Removes warning messaging on Solaris 10. We already don't support Solaris 10 in chef-land so this isn't a real loss

### CODE REVIEW

**Comment by iennae** [comment]
LGTM 👍

File: resources/build_essential.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+action :install do
+  case node['platform_family']
+  when 'debian'
+    package %w( autoconf binutils-doc bison build-essential flex gettext ncurses-dev )
+  when 'fedora', 'rhel'
+    package %w( autoconf bison flex gcc gcc-c++ gettext kernel-devel make m4 ncurses-devel patch )
+
+    # Ensure GCC 4 is available on older pre-6 EL
+    package %w( gcc44 gcc44-c++ ) if node['platform_version'].to_i < 6
```
Comment (lamont-granquist) [inline resources/build_essential.rb]: so the other pattern here to use is that you can:

```
provides :build_essential_debian
provides :build_esential, platform_family: debian
```

```
provides :build_essential_rhel
provides :build_essential, platform_family: :rhel
```

that lets someone force the selection of the resource and bypass the platform_family case statement, so its more flexible for the end user, but creates a lot of little resources...

**Comment by lamont-granquist** [comment]
think this looks fine, i threw out the comment mostly for conversation's sake...

👍

File: resources/build_essential.rb (hunk 1 above)
Comment (tas50) [inline resources/build_essential.rb]: That's the pattern I've been taking in other places, but I think there's less need / desire to do overrides here.  This is a pretty simple cookbook and if you need something funky you're probably better off just writing your own 1 line package resource.

**Pull request merged**